docs/learn/documentation/versioned/azure/eventhubs/consumer.md

```diff
@@ -0,0 +1,82 @@
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+
+You can configure your Samza job to consume from Eventhub files. The [EventhubSystemConsumer](https://github.com/apache/samza/blob/master/samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java) can read from Eventhub streams. Each Eventhub entity is read to a different Samza stream. Each partition in an Eventhub entity will naturally be mapped to a partition of the same mapped stream in Samza.
+
+### Event format
+
+The incoming message is parsed from the [EventData](https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.eventhubs._event_data). The key of the message is set to the partition key of the EventData. If that is not present, then it will be set to the user defined `key` property map of the EventData (Symmetrical to the EventhubSystemProducer). The message is obtained from the EventData body.
+
+[EventhubSystemConsumer](https://github.com/apache/samza/blob/master/samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java) receives [EventHubIncomingMessageEnvelope](https://github.com/apache/samza/blob/master/samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubIncomingMessageEnvelope.java), a special version of [IncomingMessageEnvelope](../../api/javadocs/org/apache/samza/system/IncomingMessageEnvelope.html) which contains [EventData](https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.eventhubs._event_data) that is provided by the Eventhub. The EventData allows the retrieval of extra system and user properties attached to the underlying AMQP message.
+
+
+### Consumer groups
+
+Consumer group of the stream can be configured on a stream level. EventHub documentation indicates that there can be at most [5 concurrent readers](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features#event-consumers) on a partition per consumer group. The number of available consumer groups available depends on the version of Azure subscription. Defaults to the `$Default` consumer group that exists for every stream (unless deleted). All the consumers in the same group share the upstream messages ie. the intersection of messages consumed between streams in the same group is always empty.
```

Comment:
Refer my prior comments on reorganizing this section. Also, instead of defining what consumer groups are we should just link to the eventhubs documentation.